pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -47,20 +47,37 @@ type AzVolumeSpec struct {
 	MaxMountReplicaCount int    `json:"maxMountReplicaCount"`
 	//The capabilities that the volume MUST have
 	//+optional
-	VolumeCapability *VolumeCapability `json:"volumeCapability,omitempty"`
+	VolumeCapability []*VolumeCapability `json:"volumeCapability,omitempty"`
 	//The capacity of the storage
 	//+optional
 	CapacityRange *CapacityRange `json:"capacityRange,omitempty"`
 	//Parameters for the volume
 	//+optional
 	Parameters map[string]string `json:"parameters,omitempty"`
+	//Secrets for the volume
+	//+optional
+	Secrets map[string]string `json:"secrets,omitempty"`
+	//ContentVolumeSource for the volume
+	//+optional
+	ContentVolumeSource *ContentVolumeSource `json:"contentVolumeSource,omitempty"`
+	//Specifies where the provisioned volume should be accessible
+	//+optional
+	AccessibilityRequirements *TopologyRequirement `json:"accessibilityRequirements"`
 }
 
 // AzVolumeStatus is the status for an AzVolume resource
 type AzVolumeStatus struct {
-	//Current state of the AzVolume
-	//+optional
-	State string `json:"state,omitempty"`
+	//Current status of the AzVolume
+	ResponseObject *AzVolumeStatusParams `json:"status"`
+}
+
+type AzVolumeStatusParams struct {
```

Comment:
There should also be `Status` and `StatusMessage` fields (possibly others) so that errors during disk creation can be communicated back to the controller.

Reply:
As we discussed offline, we will address this in a subsequent PR more holistically.